Module 3/SQL_Assignment.txt

```diff
@@ -0,0 +1,1880 @@
+1) List all the columns of the Salespeople table. 
+
+mysql> DESC salespeople;
++-------+-------------+------+-----+---------+-------+
+| Field | Type        | Null | Key | Default | Extra |
++-------+-------------+------+-----+---------+-------+
+| snum  | int         | NO   | PRI | NULL    |       |
+| sname | varchar(30) | NO   |     | NULL    |       |
+| city  | varchar(30) | NO   |     | NULL    |       |
+| comm  | int         | YES  |     | NULL    |       |
++-------+-------------+------+-----+---------+-------+
+4 rows in set (0.01 sec)
+
+
+2) List all customers with a rating of 100. 
+
+mysql> SELECT cname FROM customers WHERE rating = 100;
++----------+
+| cname    |
++----------+
+| Hoffman  |
+| Clemens  |
+| Pererira |
++----------+
+3 rows in set (0.00 sec)
+
+
+3) Find the largest order taken by each salesperson on each date. 
+
+mysql> SELECT s.snum, s.sname, o.odate
+    -> FROM salespeople AS s
+    -> JOIN orders AS o ON o.snum = s.snum
+    -> GROUP BY o.snum, o.odate;
++------+---------+----------+
+| snum | sname   | odate    |
++------+---------+----------+
+| 1007 | Rifkin  | 10/03/90 |
+| 1004 | Motika  | 10/03/90 |
+| 1001 | Peel    | 10/03/90 |
+| 1002 | Serres  | 10/03/90 |
+| 1002 | Serres  | 10/04/90 |
+| 1001 | Peel    | 10/05/90 |
+| 1003 | AxelRod | 10/04/90 |
+| 1002 | Serres  | 10/06/90 |
+| 1001 | Peel    | 10/06/90 |
++------+---------+----------+
+9 rows in set (0.04 sec)
+
+
+4) Arrange the Orders table by descending customer number. 
+
+mysql> SELECT * FROM orders ORDER BY cnum;
++------+---------+----------+------+------+
+| onum | amt     | odate    | cnum | snum |
++------+---------+----------+------+------+
+| 3003 |  767.19 | 10/03/90 | 2001 | 1001 |
+| 3009 | 1713.23 | 10/04/90 | 2002 | 1003 |
+| 3005 | 5160.45 | 10/03/90 | 2003 | 1002 |
+| 3007 |   75.75 | 10/04/90 | 2004 | 1002 |
+| 3010 | 1309.95 | 10/06/90 | 2004 | 1002 |
+| 3008 |    4723 | 10/05/90 | 2005 | 1001 |
+| 3011 | 9891.88 | 10/06/90 | 2006 | 1001 |
+| 3002 |  1900.1 | 10/03/90 | 2007 | 1004 |
+| 3001 |   18.69 | 10/03/90 | 2008 | 1007 |
+| 3006 | 1096.16 | 10/03/90 | 2008 | 1007 |
++------+---------+----------+------+------+
+10 rows in set (0.00 sec)
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> SELECT salespeople.sname
```

Comment:
You can have more details like snum in the output as well.

Reply:
Yes ma'am I will update the changes.